tests/patterns/kotlin/string_metavar.kt

```diff
@@ -0,0 +1,12 @@
+
+// not this because it has no part of it thats not the interpolation
+val x = "$X"
```

Comment:
Being unfamiliar with Kotlin, I was confused by these tests. Didn't realize that `$var` was the syntax for interpolation. It would help me if the test was instead `var` and `$var` or similar instead of reusing `$X`

Reply:
actually, this one is intended, because the pattern has to look like a metavariable, so its intentionally supposed to be confusing (because its solving a confusing problem)